src/main/java/optimus/Parser.java

```diff
@@ -0,0 +1,104 @@
+            String[] parts = text.split(" ");
+            if (parts.length < 2 || parts[1].trim().isEmpty()) {
+                throw new OptimusException("The task number provided is empty." +
+                        " Please provide a valid task number.");
+            }
+            int taskNumber = Integer.parseInt(parts[1]) - 1;
+            if (taskNumber < 0 || taskNumber >= record.sizeOfRecord()) {
+                throw new OptimusException("The task number provided is out of range. " +
+                        "Please provide a valid task number.");
+            }
+            Task taskToMark = record.getTask(taskNumber);
+            taskToMark.setDone();
+            ui.TaskMarked(taskToMark);
+        } else if (text.startsWith("unmark")) {
+            String[] parts = text.split(" ");
+            if (parts.length < 2 || parts[1].trim().isEmpty()) {
+                throw new OptimusException("The task number provided is empty." +
+                        " Please provide a valid task number.");
+            }
+            int taskNumber = Integer.parseInt(parts[1]) - 1;
+            if (taskNumber < 0 || taskNumber >= record.sizeOfRecord()) {
+                throw new OptimusException("The task number provided is out of range. " +
+                        "Please provide a valid task number.");
+            }
+            Task taskToUnmark = record.getTask(taskNumber);
+            taskToUnmark.setNotDone();
+            ui.TaskUnmarked(taskToUnmark);
+        } else if (text.startsWith("todo ")) {
+            String description = text.substring(5).trim();
+            if (description.isEmpty()) {
+                throw new OptimusException("The description of a todo cannot be empty. " +
+                        "Please provide a task description.");
+            }
+            return new ToDos(description);
+        } else if (text.startsWith("deadline ")) {
+            String[] parts = text.split(" /by ");
+            if (parts.length < 2) {
+                throw new OptimusException("The deadline format is incorrect. " +
+                        "Please provide a description and a deadline (e.g., deadline return book /by 2024-09-10).");
+            }
+            String description = parts[0].substring(9).trim();
+            String by = parts[1].trim();
+            if (description.isEmpty()) {
+                throw new OptimusException("The description of a deadline cannot be empty. " +
+                        "Please provide a task description.");
+            }
+            return new Deadlines(description, by);
+        } else if (text.startsWith("event ")) {
+            String[] parts = text.split(" /from | /to ");
+            if (parts.length < 3) {
+                throw new OptimusException("The event time format is incorrect. " +
+                        "Please provide both 'from' and 'to' times (e.g., event meeting /from 2024-  " +
+                        "2024-09-10 14:00 /to 2024-09-10 16:00).");
```

Comment:
I like how your error messages are very detailed and gives the user the correct format for inputs! 😄